src/convert-doi-links-to-pub-id-elements.xsl

```diff
@@ -10,7 +10,7 @@
     </xsl:copy>
   </xsl:template>
 
-  <xsl:template match="ext-link[@ext-link-type='uri'][starts-with(lower-case(@xlink:href), 'https://doi.org/')]">
+  <xsl:template match="ref//ext-link[@ext-link-type='uri'][starts-with(lower-case(@xlink:href), 'https://doi.org/')]">
     <xsl:element name="pub-id">
       <xsl:attribute name="pub-id-type">doi</xsl:attribute>
       <xsl:value-of select="substring(@xlink:href, string-length('https://doi.org/')+1)"/>
```

Comment:
I suggest 3 changes:
1. There are various ways to link using a DOI, and this only catches one variant. A more robust solution would be to use a regex (`matches` syntax), that can also capture `http` and the still relatively common (but not preferred by Crossref) `https://dx.doi.org/` convention. 

2. In the `value-of`, instead of using `substring`, `substring-after` would be more readable. 

3. Since we're using `lower-case` in the `match` expression, we should also use `lower-case` in the `value-of`, otherwise a (unlikely in real world, but possible) case where different casing is used for `doi.org` the value returned would be nothing. 

So, I suggest we change to something like:
```xml
<xsl:template match="ref//ext-link[@ext-link-type='uri'][matches(lower-case(@xlink:href),'^https?://(dx\.)?doi\.org/')]">
    <xsl:element name="pub-id">
        <xsl:attribute name="pub-id-type">doi</xsl:attribute>
        <xsl:value-of select="substring-after(lower-case(@xlink:href),'doi.org/')"/>
    </xsl:element>
</xsl:template>
```

The result will lower-case anything that ins't already lowercase, but given it's a DOI the impact is non-substantive.

Reply:
I will implement this change.

Reply:
Would be better if we could expand the test coverage to include another example of an ext-link that can expected with this new regex match.

Reply:
There's an example in 10.1101/2022.12.06.519376 - see [here](https://github.com/elifesciences/enhanced-preprints-data/blob/b3c5a1bbe43358eda8035de896ebeaef5cc598e2/data/10.1101/2022.12.06.519376/2022.12.06.519376.xml#L273):
```xml
<ext-link ext-link-type="uri" xlink:href="http://dx.doi.org/10.6080/K0G15XS1">http://dx.doi.org/10.6080/K0G15XS1</ext-link>
```

Do you want this added to the kitchen sink?